cookbooks/inner_core_convection/inner_core_convection.cc

```diff
@@ -60,6 +103,68 @@ namespace aspect
 {
   namespace MaterialModel
   {
+    template <int dim>
+    InnerCore<dim>::InnerCore ()
+      :
+      resistance_to_phase_change (1),
+      hydrostatic_pressure_profile (1)
+    {}
+
+    template <int dim>
+    double
+    InnerCore<dim>::
+    phase_function (const Point<dim> &position,
+                    const double temperature) const
+    {
+      // We need to convert the depth to pressure,
+      // keeping in mind that for this model, the pressure is only the dynamic pressure,
+      // so we have to get the hydrostatic pressure explicitly as an input.
+      const double radius = this->get_geometry_model().maximal_depth() - this->get_geometry_model().depth(position);
+      Assert(radius >= 0,
+             ExcMessage("There is a point in the model where the depth is larger "
+                        "than the maximal depth of this geometry."));
+
+      const double pressure_deviation = hydrostatic_pressure(radius) - hydrostatic_pressure(transition_radius);
+
+      double depth_deviation = transition_radius - radius;
+      if (std::fabs(pressure_deviation) > 100.0 * std::numeric_limits<double>::epsilon())
```

Comment:
I think I should have explained better. If your pressure is, say, 100GPa, i.e., 1e11, then the smallest difference you can compute in line 127 is 1e-16*1e11=1e-5. This is *always* bigger than 100*eps.

To make this `if` statement meaningful, you need to make the tolerance *relative* to the size of quantities you are comparing. So maybe something like
```
if (std::fabs(pressure_deviation) 
   > 
   100.0 * std::numeric_limits<double>::epsilon() * 
   (hydrostatic_pressure(radius) + hydrostatic_pressure(transition_radius))/2)
```
i.e., 100*eps times the *average* of the two pressures.

Reply:
I see your point, and I've changed the line accordingly. 
However, keep in mind that this model is nondimensionalized, and that the pressure is only the dynamic pressure. So the pressures are all very small, and might as well become negative, and the only thing I want to make sure is that we do not divide by zero in the next line.

Reply:
Ah, I hadn't realized that it's non-dimensional. Either way, even in non-dimensional models, quantities may be on the order of 1e-6 or 1e6, and you could run into the same problem.

But the new version looks good.